arch/arm/core/aarch32/mpu/arm_mpu.c

```diff
@@ -161,11 +162,15 @@ void arm_core_mpu_disable(void)
 {
 	uint32_t val;
 
+	/* Force any outstanding transfers to complete before disabling MPU */
+	__DMB();
```

Comment:
should this be `dsb` since `dmb` in itself does not ensure the completion of "any outstanding transfers"?

Reply:
You are right, in armv8-r `dsb` ensures the completion of the memory accesses. I was misguided looking into the armv8-m [implementation](https://github.com/zephyrproject-rtos/zephyr/commit/828ae6b8bc7169d13a3c9a04ebb6c124fa2bbd45) and the fact that armv8-r also has PMSAv8 (PMSAv8-32).